README.md

```diff
@@ -344,16 +344,21 @@ struct HeartRateMeasurement: Receivable {
         return isMeasurementIn8bits ? Int(measurement8bits) : Int(measurement16bits)
     }
 
-    init(bluetoothData: Data) {
-        flags = bluetoothData.extract(start: 0, length: 1)
-
-        isMeasurementIn8bits = (flags & 0b00000001) == 0b00000000
-
-        if isMeasurementIn8bits {
-            measurement8bits = bluetoothData.extract(start: 1, length: 1)
+    init(bluetoothData: Data) throws {
+        do {
+            flags = try bluetoothData.extract(start: 0, length: 1)
+            
+            isMeasurementIn8bits = (flags & 0b00000001) == 0b00000000
+            
+            if isMeasurementIn8bits {
+                measurement8bits = try bluetoothData.extract(start: 1, length: 1)
+            }
+            else {
+                measurement16bits = try bluetoothData.extract(start: 1, length: 2)
+            }
         }
-        else {
-            measurement16bits = bluetoothData.extract(start: 1, length: 2)
+        catch {
+            throw error
         }
```

Comment:
Not a huge deal, but if all we are doing is dethroning the error, you don't actually need the try/catch block. Errors will automatically propagate out of a throws function if uncaught, so this:
```
func a() throws {
    do {
        try something()
    } catch {
        throw error
    }
}
```
should be identical behaviour to just:
```
func a() throws {
        try something()
}
```

Reply:
Both the example here and the actual code in the demo project do that.

Reply:
Oops, I've removed the extraneous do catch clause now. Thanks!